ParseUI/Classes/QueryTableViewController/PFQueryTableViewController.h

```diff
@@ -183,6 +183,28 @@ NS_ASSUME_NONNULL_BEGIN
 - (void)removeObjectsAtIndexPaths:(nullable NSArray *)indexes animated:(BOOL)animated;
 
 /*!
+ @abstract Reflects that there's been a change in the results of `queryForTable` and there is about to be a change in `tableView` as rendered, such that the object at `_mutableObjects[indexPathSource.row]` must be moved to `_mutableObjects[indexPathDestination.row]`.
+ */
+- (void)willMoveRowForObjectFromIndexPath:(NSIndexPath *)indexPathSource toIndexPath:(NSIndexPath *)indexPathDestination;
+
+/*!
+ @abstract Reflects that there's been a change in the results of `queryForTable`, such that `tableView` as rendered must change -- specifically, the object at `indexPathSource` must now appear at `indexPathDestination` -- and the object at `_mutableObjects[indexPathSource.row]` must be moved to `_mutableObjects[indexPathDestination.row]`
+ */
+- (void)moveRowForObjectFromIndexPath:(NSIndexPath *)indexPathSource toIndexPath:(NSIndexPath *)indexPathDestination;
+
+/*!
+ @abstract Reflects that there's been a change in the results of `queryForTable`, so that `tableView` as rendered must change -- specifically, `object` must be inserted at `indexPath` -- and `_mutableObjects` must have `object` inserted at `indexPath.row`.
+ */
+- (void)insertRowForObject:(PFObject *)object atIndexPath:(NSIndexPath *)indexPath withRowAnimation:(UITableViewRowAnimation)animation;
```

Comment:
To simplify the API and remove pollution of namespace, can we remove this one in favor of the batch method at all times? It's super easy to construct arrays if you are adding only a single object, as we have literal syntax. :)